### ISSUE TITLE
Implement some form of processing queue

### ISSUE THREAD

**Issue by Skelmis** [Skelmis/DPY-Anti-Spam#13]
If a user is truely spamming, they can have a case where they incur multiple punishments due to concurrent processing of punishments. It may be the case a per user queue is implemented to ensure punishments aren't stacked

See this for an example
https://gyazo.com/dd92087470adc9fbce1a9142872db3a8

**Comment by Skelmis** [comment]
Just a thought on this, it could be implemented using the already existing message creation timestamp

**Comment by 11Tuvork28** [comment]
I made a solution, maybe not the best but it works for me :D 
All changes made in User.py

Line 31:
Added:
```
slot oldDuplicateCounter, kicked
```
Line 58:
Added:
```
self.oldDuplicateCounter = 0 //Needs to be zero
self.kicked = False 
``` 
If the user gets kicked it gets set to True in the kick function and when the user rejoins it gets set to False again.
Line 121:
Added:
```
if self.kicked:
     return
```
We don't need to do anything if the user got kicked
Line 155:
Added:
``` 
if self.oldCounter != 0:
      self.duplicateCounter = self.oldDuplicateCounter
```
This restores the old value of duplicateCounter.
Line 186:
Added:
```
self.oldDuplicateCounter = self.duplicateCounter
self.duplicateCounter = 0
```
This sets the oldDuplicateCounter and resets the duplicateCounter to zero this time because we would add 2 otherwise on line 157(156 without the changes I mentioned).

With this tiny code two problems get fixed:
- The bug you mentioned
- The ability that a user can spam and gets warned but can write normal again if the user stops spamming. This wouldn't be possible otherwise ? Or am I wrong.

What do you think?

**Comment by Skelmis** [comment]
I like the process behind it, however, I have a couple of issues / questions regarding your implementation.
1. How are you setting if the user has rejoined the guild or not
2. Can you create a pull request with this code so that I can test it, also mention this issue in your description for it
3. I personally think there might be a slightly better way, given it also relates all punishments, I think? But its a step in the right direction

**Comment by Skelmis** [comment]
In relation to your second point about how after time a user should be able to write normally again after x time (How I am interpreting it), this is now resolved with Commit a493d17d56e272e0a6ca7e6c8c9aef2663ca8761

The idea behind this is now after the default timeout period, any messages older than that are deleted and are no longer relevant/related to a user 'spamming' since messages and spam is defined as within a time period. Before this change, you could say something today, then again tomorrow and its counted as spam.
